ui/phase_manager.py

```
@@ -0,0 +1,344 @@
+            self._phase_steps[phase_id] = []
+
+        phase = self._get_phase_info(phase_id, name)
+
+        self._queue.publish({
+            "type": "phase_start",
+            "phase": phase,
+            "progress": self._get_progress(),
+            "queue": self._get_pending(),
+        })
+
+    def done_phase(self, phase_id: str, name: str = None) -> None:
```

Ruff (0.15.9)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)